doc/OSAL-Configuration-Guide.md

```diff
@@ -499,360 +450,263 @@ Using the host vxWorks tools, the path to the same file would be:
+in a black box test enviornment.
+
+These test programs are executed just as any other application program.
+
+Complete coverage testing is only available for the "shared" layer and
+the "VxWorks" low level implementation layer.  However, because all platforms
+share the same "shared" layer, this coverage test applies to all platforms.
+The low level (OS-specific) implementation layers should be only to translate
+the OSAL calls to the specific C library/OS API provided by that platform,
+while all basic state/table maintenance and error checking is performed at
+the shared layer.
+
+**Note**: The coverage tests are built for the same platform as all other
+applications.  Notably, because the coverage test does not actually use
+the facilities provided by the OS (all OS calls are replaced with stubs),
+the OS-specific coverage test can execute on any platform, it is not limited
+to running on the platform that it is abstracting.  As a result, the VxWorks
+coverage test is typically executed on a Linux host.
 
 ## How to Run the OSAL Unit Tests on the PC / Linux Platform
 
 To build and run the OSAL unit tests, run the following commands from
-the "build" directory: 
+the "build" directory:
 
 #### Build Commands
 
 |  **Shell command**      |    **Description**|
 |---|---|
-|  **\$ cd build**|Change to the build directory.|
-| **\$ make clean-unit-tests**|   Clean the OSAL Core, and all Applications|   
-|**\$ make config**|Copy the osconfig.h for the BSP to the build directory|  
-| **\$ make unit-tests**|Build all of the unit tests.| 
-|  **\$ make gcov**|Run the unit tests, collecting code coverage information.| 
+|  `$ cd build/unit-test-coverage` | Change to the build directory containing coverage tests |
+| `$ make` | Build the OSAL Core, and all tests |
+| `$ make test` | Execute all of the tests programs |
 
-Once the unit tests have run, the log files are available in the
-individual unit test binary directories 
+Note that the "ctest" program shows a quiet output by default, indicating only the test name
+and a pass/fail status.  To see a complete log of tests executed, the test program may be run
+directly from the command line.
 
 #### OSAL Unit Test Directories
 
  | **Directory**|**Description**|
  |---|---|
- |  build|The top level OSAL build directory.|
- |  build/unit-tests|The top level unit test build directory.|
- |  build/unit-tests/oscore-test|Contains the test binary, log, and gcov files for the OSAL core tests|  |  build/unit-tests/osfile-test|Contains the test binary, log, and gcov files for the OSAL file API tests|  
- | build/unit-tests/osfilesys-test|Contains the test binary, log, and gcov files for the OSAL file system tests|  
- | build/unit-tests/osloader-test|Contains the test binary, log, and gcov files for the OSAL loader tests|  
- |build/unit-tests/ostimer-test|Contains the test binary, log, and gcov files for the OSAL timer tests| 
+ | `build` | The top level OSAL build directory. |
```

Comment:
Is this intentionally excluding the coverage tests, or should it be included?  Might be worth detailing all three test groups are built and run (w/ a native setup) via the sample makefile., and can cross-compile and build everything to run on a target (may be covered above, but might be worth repeating in the testing section?)

Reply:
I removed this section entirely because it doesn't really add value (I think the subdirectories under unit-tests are pretty self-explanatory) and there was already a list of directories earlier in the document.  

I added an entry for `unit-test-coverage` there (the document was originally written before this existed).